src/m365/tenant/commands/service/service-list.ts

```diff
@@ -0,0 +1,69 @@
+import commands from '../../commands';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+class TenantServiceListCommand extends SpoCommand {
+  public get name(): string {
+    return `${commands.TENANT_SERVICE_LIST}`;
+  }
+
+  public get description(): string {
+    return 'Gets the services available in Microsoft 365 from the Microsoft 365 Management API';
+  }
+
+  public commandAction(cmd: CommandInstance, args: any, cb: (err?: any) => void): void {
+    if (this.verbose) {
+      cmd.log(`Getting the health status of the different services in Microsoft 365.`);
+    }
+
+    const serviceUrl: string = 'https://manage.office.com/api/v1.0';
+    const statusEndpoint: string = 'ServiceComms/Services';
+
+    this
+      .getTenantId(cmd, this.debug)
```

Comment:
I'd suggest that we use a different approach to get the tenant ID. Using the `SpoCommand.getTenantId` method requires the user to have access to the SharePoint admin site, which shouldn't be necessary if all you're interested is service information. What we can do instead is to write a helper method named `getTenantIdFromAccessToken` similar to `Utils.getUserNameFromAccessToken`, which would extract the tenant ID from the access token. If you look at the `Utils.getUserNameFromAccessToken` method, it gets the user name from the `token.upn` property. Similarly, tenant ID is stored in the `token.tid` property. Not only it doesn't require any extra permissions but it also saves us a server roundtrip. With this change, we could extend the `Command` class rather than `SpoCommand` which is specific to SharePoint.

Reply:
Thanks for the suggestion and the detailed explanation @waldekmastykarz. Do you recommend adding this helper method to this specific `TenantServiceListCommand` class or the `Utils` class?

Reply:
Let's add it to the `Utils` class so that we can use it across the CLI.

Reply:
Hi @aakashbhardwaj619, @waldekmastykarz 
I have added `Utils.getTenantIdFromAccessToken`
https://github.com/pnp/cli-microsoft365/pull/1781

Reply:
#1781 is now merged so if you'd rebase your fork, you should have it available in the `Utils` class @aakashbhardwaj619

Reply:
Hi @waldekmastykarz, sorry for the late response. I got caught up with some other tasks, but I've now rebased my fork with the latest changes. However, the test case coverage has dropped for the new `getTenantIdFromAccessToken` method. Not sure why it has happened for this PR but did not happen for #1781 since the `Utils` class is the same. Could you please advise?

Reply:
Could it be that your rebase didn't go as intended and something got removed in the process?

Reply:
Maybe, I'm going to create a new pull request that contains the updated rebased code and should work fine